adafruit_max31855.py

```diff
@@ -0,0 +1,75 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2017 Radomir Dopieralski for Adafruit Industries.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import ustruct
+
+
+class MAX31855:
+    """
+    Driver for the MAX31855 thermocouple amplifier.
+    """
+
+    def __init__(self, spi, cs):
+        self.spi = spi
+        self.cs = cs
+        self.data = bytearray(4)
+
+    def read(self, internal=False, raw=False):
+        """
+        Read the measured temperature.
+
+        If ``internal`` is ``True``, return a tuple with the measured
+        temperature first and the internal reference temperature second.
+
+        If ``raw`` is ``True``, return the values as 14- and 12- bit integers,
+        otherwise convert them to Celsuius degrees and return as floating point
+        numbers.
+        """
```

Comment:
I considered other approaches:

- Having two separate methods (or properties) that would read all the data, but discard half of it and only return the other half. This is useless, as you really want to get the internal reference temperature from exactly the same measurement, as your temperature measurement. Doing two separate measurements defeats the whole purpose.
- Returning the temperature, and saving the internal reference to an attribute on this class. This introduces an asymmetry, where one property causes a new measurement to be made, but the other doesn't. I found that really confusing and surprising, so I decided against it.
- The problem with the former approach could be somewhat alleviated by naming the attribute appropriately, such as "last_measurement_internal_reference_temperature" or something similar, however that gets ridiculously long.